pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,377 @@
+        get_process=lambda err: DPSeriesErrors(int(err)),
+    )
+
+    def check_errors(self):
+        """ Method to read the error codes register and log when an error is detected.
+        
+        :return error_code: one byte with the error codes register contents
+        """
+        current_errors = self.error_reg
+        if current_errors != 0:
+            logging.error("DP-Series motor controller error detected: %s" % current_errors)
+        return current_errors
+    
+    def __init__(self, resourceName, address=0, encoder_enabled=False, **kwargs):
+        """
+        Initialize communication with the motor controller with id=idn. In addition to the keyword arguments that can be
+        set for the Instrument base class, this class has the following kwargs:
+
+        :param address: (int) Address that the motor controller uses for serial communiations.
+        :param encoder_enabled: (bool) Flag to indicate if an encoder has been connected to the controller. 
+        """
+        self._address = address
+        self._encoder_enabled = encoder_enabled 
+
+        super().__init__(
+            resourceName,
+            "Anaheim Automation Stepper Motor Controller",
+            includeSCPI=False,
+            write_termination="\r",
+            read_termination="\r",
+            timeout=2000, 
+            **kwargs
+        )
+        
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.connection.baud_rate = 38400
+    
+    @property
+    def encoder_enabled(self):
+        """
+        Return the value of the _encoder_enabled flag.
+        """
+        return self._encoder_enabled
+
+    @encoder_enabled.setter
+    def encoder_enabled(self, en):
+        """ Set the value of the _encoder_enabled flag. When set, asynchronous coroutine methods like step_async() will
+            query the "encoder_position" property instead of the "position" property.
+
+        :param en: (bool) boolean value to set the _encoder_enabled flag with
+        """
+        self._encoder_enabled = bool(en)
+    
+    @property
+    def step_position(self):
+        """
+        Return the value of the motor position measured in steps counted by the motor controller or, if encoder_enabled
+        is set, return the steps counted by an externally connected encoder.
+        """
+        if self._encoder_enabled:
+            pos = self.ask("VEP")
+        else:
+            pos = self.ask("VZ")    
+        return int(pos)
+    
+    @step_position.setter
+    def step_position(self, pos):
+        """ Sends command to the motor controller to move to the desired step position. Note that in DP series
+            controller instrument manuals, this property corresponds to the 'absolute position' command. In this driver,
+            the `absolute_position` property implements a conversion between steps to absolute units whereas the DP
+            series manuals refer to `absolute position` as measured in steps.
```

Comment:
great that you explain this in detail, for clarity! Watch out, though, Python properties only get one docstring, I think it's the one from the getter? Best combine them into one. You can check with `myproperty._doc__`, or in IPython `myproperty?`.